Ninja: reformat expression

AndersonTorres opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

AndersonTorres committed:
ninja: add runHook's

AndersonTorres committed:
ninja: format expression

Because I believe it is nicer this way.

AndersonTorres commented:
Why the failure?

SuperSandro2000 commented on pkgs/development/tools/build-managers/ninja/default.nix:
```diff
@@ -1,39 +1,67 @@
     repo = "ninja";
-    rev = "v${version}";
-    sha256 = "sha256-LvV/Fi2ARXBkfyA1paCRmLUwCh/rTyz+tGMg2/qEepI=";
+    rev = "v${finalAttrs.version}";
```
```suggestion
    rev = "v${version}";
```
that is not working because of the fixed output derivation

AndersonTorres replied:
@roberth ping as a new bad case

roberth replied:
> not working

You _can_ replace `rec` like this.

I suppose you expect to be able to override both `rev` and `hash` in one go? `finalAttrs` would make this work if `fetchFromGitHub` had a `.override`:

```nix
(ninja.overrideAttrs(_: { version = "9000"; })
).overrideAttrs(old: {
  src = super.src.override (_: { hash = "foo"; });
})
```

It's DRY in terms of the packaging domain, but overriding the "same" derivation twice is a little excessive.

This could be prevented by adding `hash` to the `ninja` derivation directly, but that derivation shouldn't have `hash`, as the `ninja` builder doesn't (and probably shouldn't) use it.
For it to work in one `overrideAttrs`, or perhaps one `overrideAttrs` per derivation (`ninja`, `source`), we'd need further changes, and I don't have a single answer for what those should be.

While we might do slightly better than the status quo, overriding is never going to be really pretty, because it is fundamentally coupled with the implementation of whatever you're overriding.

AndersonTorres replied:
I am being more modest here and merely getting rid of `rec`.

I will try another time.

SuperSandro2000 commented on pkgs/development/tools/build-managers/ninja/default.nix:
```diff
@@ -1,39 +1,67 @@
-with lib;
-
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
```
```suggestion
stdenv.mkDerivation rec {
```

AndersonTorres committed:
ninja: use installShellCompletion

AndersonTorres committed:
ninja: revert to the old rec format

AndersonTorres committed:
ninja: reorder setup-hook.sh

It is considerably more readable to put the functions-hooks before the code
activating them.

AndersonTorres commented:
I will rebase and try again.

mweinelt commented on pkgs/development/tools/build-managers/ninja/default.nix:
```diff
@@ -10,30 +20,50 @@ stdenv.mkDerivation rec {
-    install -Dm444 misc/bash-completion $out/share/bash-completion/completions/ninja
-    install -Dm444 misc/zsh-completion $out/share/zsh/site-functions/_ninja
-  '' + optionalString buildDocs ''
+    installShellCompletion --name ninja --bash misc/bash-completion
+    installShellCompletion --name ninja --zsh misc/zsh-completion
```
```suggestion
    installShellCompletion --name ninja \
       --bash misc/bash-completion \
       --zsh misc/zsh-completion
```

AndersonTorres replied:
Cool! I will commit this.

AndersonTorres commented:
please
@ofborg eval

AndersonTorres commented:
Now this is "morally equivalent" to the previous one.
